pytroll_collectors/s3stalker_daemon_runner.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2020 - 2023 Pytroll developers
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""S3stalker daemon.
+
+The contents of the yaml configuration file should look like this::
+
+    s3_kwargs:
+      anon: false
+      client_kwargs:
+        aws_access_key_id: my_accesskey
+        aws_secret_access_key: my_secret_key
+        endpoint_url: https://xxx.yyy.zz
+    fetch_back_to:
+      hours: 20
+    polling_interval:
+      minutes: 2
+    file_pattern: '{platform_name:3s}_OL_2_{datatype_id:_<6s}_{start_time:%Y%m%dT%H%M%S}_{end_time:%Y%m%dT%H%M%S}_{creation_time:%Y%m%dT%H%M%S}_{duration:4d}_{cycle:3d}_{relative_orbit:3d}_{frame:4d}_{centre:3s}_{mode:1s}_{timeliness:2s}_{collection:3s}.zip'
+    subject: /segment/2/safe-olci/S3/
+"""  # noqa
+import signal
+import time
+from datetime import timedelta, datetime
+from threading import Thread
+
+from dateutil.tz import UTC
+
+from posttroll.publisher import create_publisher_from_dict_config
+
+from pytroll_collectors.s3stalker import logger, create_messages_for_recent_files, set_last_fetch, sleeper
+
+
+class S3StalkerRunner(Thread):
+    """Runner for stalking for new files in an S3 object store."""
+
+    def __init__(self, bucket, config, publisher_ready_time=2.5):
+        """Initialize the S3Stalker runner class."""
+        super().__init__()
+
+        self.bucket = bucket
+        startup_time = timedelta(**config.pop("fetch_back_to"))
```

Comment:
This isn't startup time, so the variable name is misleading.

The same config item is accessed in `s3stalker.py` function `publish_new_files()`. Is the daemon passing the same config dictionary to it or are these independent? If the same config is passed, the other is passing `None` to `timedelta()`.

Reply:
Fixed the name

Reply:
`publish_new_files` is not called by the daemon, since this is made to be called once and exit.